policy.go

```diff
@@ -494,28 +644,55 @@ func ensureSufficientORDigests(digests tpm2.DigestList) tpm2.DigestList {
 	return digests
 }
 
+// computeDynamicPolicyRef computes the reference used for authorization of signed dynamic authorization policies.
+func computeDynamicPolicyRef(alg tpm2.HashAlgorithmId, counterPublic *tpm2.NVPublic) (tpm2.Nonce, error) {
+	if !alg.Supported() {
+		return nil, errors.New("unsupported algorithm")
+	}
+
+	var counterName tpm2.Name
+	if counterPublic != nil {
+		var err error
+		counterName, err = counterPublic.Name()
+		if err != nil {
+			return nil, xerrors.Errorf("cannot compute name of counter: %w", err)
+		}
+	}
+
+	h := alg.NewHash()
+	h.Write([]byte("DYNAMIC-PCR"))
+	h.Write(counterName)
```

Comment:
Do we need a closing sentinel of some sort to prevent length extension attacks on this field?

Reply:
We shouldn't do - the field is always the same length (it's a digest preceded by 2 bytes to indicate the algorithm).